app/controllers/users/passwords_controller.rb

```diff
@@ -8,9 +8,19 @@ class Users::PasswordsController < Devise::PasswordsController
   # end
 
   # POST /resource/password
-  # def create
-  #   super
-  # end
+  def create
+    # Check the credentials associated to the mail to generate a correct reset link
+    email = params[:user][:email]
+    if Administrateur.find_by(email: email)
+      @devise_mapping = Devise.mappings[:administrateur]
+      params[:administrateur] = params[:user]
+      # uncomment to check password complexity for Gestionnaire
```

Comment:
👍 Est-ce que ça serait pas aussi simple d’en faire une feature dans flipflop? (peut-être pas, j’en sais rien: mais si on prévoit raisonnablement de le faire, ça pourra nous faire gagner du temps)

Reply:
Désolé, je n'ai pas compris le lien entre flipflop et le fait de sélectionner la page de changement de mot de passe en fonction du niveau de l'utilisateur (admin, instructeur, utilisateur). Peux tu m'éclairer ?

Reply:
La fonctionnalité “check password complexity for Gestionnaire” pourrait être entourée d’un feature-flag:

```
elsif Gestionnaire.find_by(email: email)
  if Flipflop.check_password_complexity_for_gestionnaire?
    @devise_mapping = Devise.mappings[:gestionnaire]
    params[:gestionnaire] = params[:user]
  end
end
```

pour permettre d’activer/désactiver la fonctionnalité depuis `/manager/features` sans (dé)commenter le code.

Reply:
Ok je comprends. Voici un peu de contexte: 
J'avais mis en place une mesure de la complexité du mot de passe pour les instructeurs. @kemenaran  considère qu'avec la 2FA mise en place récemment, la sécurité est assurée et cela permet aux usagers d'utiliser leur mot de passe habituel. Donc pas besoin de mesure de la complexité pour les instructeurs. 
J'avais mis en commentaire sur le moment pour le cas ou quelqu'un voudrait par la suite ajouter la mesure de la complexité pour les instructeurs. Je pense que vais enlever ce code commenté, je n'ai pas envie de polluer le flipflop avec une feature qui n'est pas utile pour vous.